File: d1s2.php
<?php

$array = file("input.txt");
$num_larger = 0;
for($i = 0; $i+3 < count($array); $i++){
    $groupA = (int)$array[$i] + (int)$array[$i+1] + (int)$array[$i+2];
    $groupB = (int)$array[$i+1] + (int)$array[$i+2] + (int)$array[$i+3];

    if($groupB > $groupA){
        $num_larger++;
    }
}

var_dump($num_larger);

File: d1s2.py
text_file = open("/Users/cj/IdeaProjects/adventofcode2021/d1/input.txt", "r")
lines = text_file.readlines()
text_file.close()

num_larger = 0
previous_line = None

for count in range(0,len(lines)):
    if count+3 >= len(lines):
       break;

    groupA = int(lines[count]) + int(lines[count+1]) + int(lines[count+2])
    groupB = int(lines[count+1]) + int(lines[count+2]) + int(lines[count+3])

    if groupB > groupA:
        num_larger = num_larger + 1

print(num_larger)





